api/src/main/java/org/apache/iceberg/ManifestFile.java

```diff
@@ -126,7 +126,7 @@ static Schema schema() {
   /** Returns the sequence number of the commit that added the manifest file. */
   long sequenceNumber();
 
-  /** Returns the lowest sequence number of any data file in the manifest. */
+  /** Returns the lowest data sequence number of any live file in the manifest. */
   long minSequenceNumber();
```

Comment:
I believe this is what was happening before as entries with status DELETED had null as its sequence number and it was not taken into account. No behavior change, just updating the comment.